File: frontend/lib/schemas/psa.ts
import { DocSchema } from "../doc-schema";

export const psaSchema: DocSchema = {
  id: "psa",
  title: "Professional Services Agreement",
  subtitle: "Cover Page",
  infoBox:
    'This Professional Services Agreement (the "PSA") consists of: (1) this Cover Page, (2) one or more Statements of Work (SOWs), and (3) the Common Paper Professional Services Agreement Standard Terms Version 1.0 identical to those posted at commonpaper.com/standards/professional-services-agreement/1.0. Any modifications of the Standard Terms should be made on the Cover Page, which will control over conflicts with the Standard Terms.',
  coverFields: [
    {
      key: "effectiveDate",
      label: "Effective Date",
      type: "date",
      placeholder: "Select a date",
    },
    {
      key: "governingLaw",
      label: "Governing Law",
      type: "text",
      placeholder: "State",
    },
    {
      key: "chosenCourts",
      label: "Chosen Courts",
      type: "text",
      placeholder: "City/county and state",
    },
  ],
  parties: [
    { label: "Provider", fields: ["name", "title", "company", "noticeAddress", "date"] },
    { label: "Customer", fields: ["name", "title", "company", "noticeAddress", "date"] },
  ],
  signingNote: "By signing this Cover Page, each party agrees to enter into this Professional Services Agreement as of the Effective Date. Specific project details, deliverables, and fees will be defined in separate Statements of Work.",
  templateFile: "templates/psa.md",
  footerNote: "Common Paper Professional Services Agreement (Version 1.0) free to use under CC BY 4.0.",
  greeting:
    "Hi! I'm your Professional Services Agreement assistant. I'll help you fill in the PSA Cover Page through conversation. This is the framework agreement for engaging professional services -- specific project details will be defined in separate Statements of Work. Let's start -- when should this agreement take effect? (e.g. 'Today', 'April 1 2026', 'Immediately')",
  pdfFilename: "professional-services-agreement.pdf",
};

File: backend/doc_configs/psa.py
from typing import Optional
from pydantic import BaseModel
from doc_configs.shared import PartialPartyInfo


SYSTEM_PROMPT = """You are a legal document assistant helping a user fill in a Professional Services Agreement (PSA) Cover Page.

## Fields you must collect

1. **Effective Date** -- when the agreement takes effect
   Ask: "When should this agreement take effect? (e.g. 'today', 'April 1 2026', 'next Monday')"
   For shortcuts like "today", "tomorrow", "Mon", "next Mon", use this lookup table:
{date_lookup}
   For full dates (e.g. "March 15 2026", "2026-03-15"), convert to YYYY-MM-DD format directly.

2. **Governing Law** -- jurisdiction whose laws govern the agreement
   Ask: "Which state or jurisdiction's law should govern this agreement? (e.g. 'California', 'New York', 'Delaware')"

3. **Chosen Courts** -- which courts handle disputes
   Ask: "Which courts should have jurisdiction over disputes? (e.g. 'San Francisco, California', 'New York County, New York')"

4. **Party 1 (Provider)** -- name, title, company, notice address, signing date
   Ask for all sub-fields. Example: "Jane Smith, CEO of ConsultCo, jane@consultco.com, signing today"

5. **Party 2 (Customer)** -- same sub-fields as Party 1

## Current document state
{current_fields}

## Instructions
- Work through the fields in order, but adapt naturally to what the user already told you
- Ask 1-2 questions per turn maximum
- Always include a concrete example in parentheses when asking a question
- When the user answers, confirm what you captured then always end with a follow-up question
- Note: specific project details, deliverables, and fees are defined in separate Statements of Work, not on this Cover Page
- Once all fields are collected, tell the user the document is complete and they can download the PDF

## Clearing fields
If the user asks to clear fields (e.g. "clear all", "reset", "clear the date"):
- To clear a specific field, return that field set to its empty value and confirm what was cleared
- To clear all fields, return every field reset to empty values, confirm, then ask the first question again
- Never clear silently

Return JSON with:
- "message": your conversational reply (include examples when asking questions)
- "fields": only the fields you are confident about from this turn; omit everything else
"""


class PartialFields(BaseModel):
    effectiveDate: Optional[str] = None
    governingLaw: Optional[str] = None
    chosenCourts: Optional[str] = None
    party1: Optional[PartialPartyInfo] = None
    party2: Optional[PartialPartyInfo] = None


class AiResponse(BaseModel):
    message: str
    fields: PartialFields
